### PR TITLE
Issue 577: Add backend part for orgs full-text search

### PR DESCRIPTION
**Pull request by lsenta**

### CODE REVIEW

File: apps/backend-functions/src/algolia.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import { algoliaAdminKey, algoliaId } from './environments/environment';
+
+const indexOrganizationsBuilder = () => {
+  const client = algoliasearch(algoliaId, algoliaAdminKey);
+  const INDEX_NAME_ORGANIZATIONS = 'orgs';
+  return client.initIndex(INDEX_NAME_ORGANIZATIONS);
+};
+
+export function storeSearchableOrg(orgId: string, name: string): Promise<any> {
+  if (algoliaId === "") {
```
Comment (GrandSchtroumpf) [inline apps/backend-functions/src/algolia.ts]: You should check of `!algoliaId`, like that you can check for undefined and null also.

File: apps/backend-functions/src/environments/environment.prod.ts [hunk 2]
```diff
@@ -10,3 +10,7 @@ export { factoryContract, backupBucket, relayer, appUrl } from '@env';
 
 export const sendgridAPIKey = functions.config().sendgrid.apiKey;
 export const mnemonic = functions.config().relayer.mnemonic;
+
+export const algoliaId = functions.config().algolia.app_id;
```
Comment (GrandSchtroumpf) [inline apps/backend-functions/src/environments/environment.prod.ts]: if functions.config() require ressources, I would rather store the return value and exports the keys : 
```
export const {
 app_id: algoliaId,
 api_key: algoliaAdminKey,
 search_key: algoliaSearchKey
} = functions.config().algolia;
```

File: apps/backend-functions/src/algolia.ts (hunk 1 above)
Comment (lsenta) [reply apps/backend-functions/src/algolia.ts]: That's only for dev mode when the rest of the team work with the app without an algolia account, the tighter the check the safer.

File: apps/backend-functions/src/environments/environment.prod.ts (hunk 2 above)
Comment (lsenta) [reply apps/backend-functions/src/environments/environment.prod.ts]: Its OK, its designed to be used like a process.env

**Pull request merged**